### PR TITLE
[client]Check  consumer schema null in advance

### PR DESCRIPTION
**Pull request by AnonHxy**
### Motivation

* If we set `scheam` as null when using `PulsarClient` new consumer  like this:
```
Consumer<?> consumer = client.newConsumer(null)
                .topic(topic)
                .subscriptionName(subscription)
                .subscribe();
```
It does not throw exceptions when building consumer, but will throw exceptions when `org.apache.pulsar.client.impl.ConsumerImpl#connectionOpened` invoked, then it will try to reconnect later. See the following log detail：
image

* For detail, the code where throws exception is located at： 
https://github.com/apache/pulsar/blob/74eae1a729c33c9fead9d54594bdee5fac8ab153/pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerImpl.java#L827

### Modifications

If `schema` is null, throw exceptions when buiding the consumer

### Verifying this change

- [x] Make sure that the change passes the CI checks.

### Documentation
  
- [x] `no-need-doc`

### CODE REVIEW

File: pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerBuilderImpl.java [hunk 1]
```diff
@@ -77,6 +77,7 @@ public ConsumerBuilderImpl(PulsarClientImpl client, Schema<T> schema) {
     }
 
     ConsumerBuilderImpl(PulsarClientImpl client, ConsumerConfigurationData<T> conf, Schema<T> schema) {
+        checkArgument(schema != null, "schema should not be null");
```
Comment (RobertIndie) [inline pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerBuilderImpl.java:80]:
```suggestion
        checkArgument(schema != null, "Schema should not be null.");
```

File: pulsar-client/src/test/java/org/apache/pulsar/client/impl/PulsarClientImplTest.java [hunk 2]
```diff
@@ -225,6 +226,20 @@ public void testResourceCleanup() throws PulsarClientException {
         }
     }
 
+    @Test
+    public void testConsumerSchemaNull() throws PulsarClientException {
+        @Cleanup
```
Comment (RobertIndie) [inline pulsar-client/src/test/java/org/apache/pulsar/client/impl/PulsarClientImplTest.java:231]: This doesn't seem to be part of the PulsarClientImpl test, we should move it to the consumer test.

Comment (AnonHxy) [reply pulsar-client/src/test/java/org/apache/pulsar/client/impl/PulsarClientImplTest.java]: OK. PTAL @RobertIndie

**Review by HQebupt — APPROVED** [review]

**Review by lhotari — APPROVED** [review]

**Pull request merged**